include/zephyr/kernel/internal/mm.h

```diff
@@ -56,6 +56,7 @@
 #include <stddef.h>
 #include <inttypes.h>
 #include <zephyr/sys/__assert.h>
+#include <zephyr/sys/mem_manage.h>
```

Comment:
This results in circular includes. `sys/mem_manage.h` includes `kernel/mm.h` which already includes `kernel/internal/mm.h`.

Reply:
hmm, it seems that `sys/mem_manage.h` shouldn't be including `kernel/mm.h`, since it does not need anything from there. The include guards are protecting from this circular problem, but that is not a good solution. 
Going to remove "kernel/mm.h" from "sys/mem_manage.h".